README.md

```diff
@@ -1,2 +1,189 @@
+```ballerina
+type Ticket record {|
+    int id;
+    Restaurant restaurant;
+    Order 'order;
+    TicketState status;
+|};
+```
+
+A ticket can be in one of the following states:
+* `ACCEPTED`
+* `PREPARING`
+* `READY_FOR_PICKUP`
+* `PICKED_UP`
+
+When a ticket is created, it will initially be in the `ACCEPTED` state. The state of a ticket can be advanced to each state using the three endpoints defined.
+* To `PREPARING`: `restaurant/<restaurantId>/ticket/<ticketId>/mark/preparing`
+* To `READY_FOR_PICKUP`: `restaurant/<restaurantId>/ticket/<ticketId>/mark/ready`
+* To `PICKED_UP`: `restaurant/<restaurantId>/ticket/<ticketId>/mark/pickedUp`
+
+In each of these scenarios, the corresponding state of the `Order` in the `Order Service` is also changed.
+
+![Ticket Sequence](/assets/ticket_sequence.png)
+
+
+### Data storage and retrieval
+The restaurant service has it's own MySQL database for storing relevant restaurant data. Since this services accesses data from the `Order Service` it is required to configure this endpoint and make REST API calls whenever necessary.
+
+### Running the module
+1. Set up a MySQL database and create the relevant tables using the queries in the `init.sql` file in the `restaurant` directory.
+2. Configure the database connection properties as well as the order endpoint in the `Config.toml` file in the `restaurant` directory. 
+3. To run the consumer service, simply navigate to the `restaurant` directory and execute `bal run` in the terminal. 
+
+## Order service
+The order service handles orders placed by a consumer for a restaurant. This service is central to this example. 
+
+The basic data structure of an order is as follows:
+```ballerina
+type Order record {|
+    int id;
+    Consumer consumer;
+    Restaurant restaurant;
+    OrderItem[] orderItems;
+    string deliveryAddress;
+    time:Civil deliveryTime;
+    OrderState status;
+|};
+
+type OrderItem record {|
+    int id;
+    MenuItem menuItem;
+    int quantity;
+|};
+```
+
+This service provides 4 endpoints to perform basic CRUD functionalities with orders.
+
+An order can be in one of the following states:
+* `APPROVAL_PENDING`
+* `APPROVED`
+* `REJECTED`
+* `ACCEPTED`
+* `PREPARING`
+* `READY_FOR_PICKUP`
+* `PICKED_UP`
+* `DELIVERED`
+* `CANCELLED`
+
+As an order is created, it is initially in the `APPROVAL_PENDING` stage. 
+- As it moves into the `ACCEPTED` stage after the order is confirmed, a ticket has to be created in the restaurant. 
+- As it moves into the `READY_FOR_PICKUP` stage, a delivery has to be scheduled using the `Delivery Service`.
+
+![Order Sequence](/assets/order_sequence.png)
+
+
+### Data storage and retrieval
+The restaurant service has it's own MySQL database for storing relevant restaurant data. Since this services accesses data from the other services it is required to configure these endpoints and make REST API calls whenever necessary.
```

Comment:
```suggestion
The restaurant service has its own MySQL database for storing relevant restaurant data. Since this service accesses data from the other services it is required to configure these endpoints and make REST API calls whenever necessary.
```